### DISCUSSION TITLE
TextEncoder is not a constructor error in nextjs server side api

### DISCUSSION THREAD

**Discussion by vycos-zen** [eleith/emailjs#333] [questions]
Hi,

I ran into an issue when implementing emailjs SMTPClient in a NextJs server side api.

the error: 
`TypeError: util__WEBPACK_IMPORTED_MODULE_3__.TextEncoder is not a constructor`

implementation:
```
import { SMTPClient } from "emailjs";

export const emailClient = () =>
  new SMTPClient({
    user: process.env.EMAILJS_USER,
    password: process.env.EMAILJS_PASSWORD,
    host: process.env.EMAILJS_HOST,
    ssl: true,
  });
```
api endpoint call:
```
  try {
    const { email } = req?.body;

    if (!email) {
      return res.status(500)?.json({
        message: "Missing email",
      });
    }

    const client = emailClient();

    const textConstruct = `bruderbau.hu kapcsolat felvétel üzenet: 
      
      ${email.text}`;

    const messageConstruct = `Üzenet ${email.from} feladótól, bruderbau.hu kapcsolat felvétel`;
    const from = email.from;

    const message = await client.sendAsync({
      text: textConstruct,
      from: from,
      to: process.env.EMAILJS_DEFAULT_RECIPENT ?? "",
      subject: messageConstruct,
    });

    return res.status(200)?.json({
      message: message.text,
    });
  } catch (error) {
    res.status(500)?.json({
      message: "Error submitting email",
    });
  }
```

call stack extract:
![Screenshot 2022-12-28 at 14 04 37](https://user-images.githubusercontent.com/8149214/209816705-902ad568-6174-4d50-9347-8888f2016894.png)

i want to release this feature to my client as soon as possible. please let me know if it`s something on my side, or more info needed. Also if no possibility of fixing it, so I look for an alternative solution.

Best regards,
Peter

**Comment by eleith** [comment]
currently, i can not reproduce (i have it successfully deployed in a nextjs application)

the error is coming from this line: https://github.com/eleith/emailjs/blob/main/email.js#L270

this is a builtin nodejs module, thus, the error suggests that nextjs is trying to embed your email setup logic for browser execution (since that is what they use webpack for, to bundle code for the browser, and this error is coming from nextjs use of webpack)

make sure all email logic is exclusively accessible by only the API route and not any nextjs pages
